processing/src/main/java/io/druid/query/QueryRunner.java

```diff
@@ -24,14 +24,27 @@
 import java.util.Map;
 
 /**
+ * This interface has two similar run() methods. {@link #run(Query, Map)} is legacy and {@link #run(QueryPlus, Map)}
+ * is the new one. Their default implementations delegate to each other. Every implementation of QueryRunner should
+ * override only one of those methods. New implementations should override the new method: {@link #run(QueryPlus, Map)}.
  */
 public interface QueryRunner<T>
 {
   /**
-   * Runs the given query and returns results in a time-ordered sequence
-   * @param query
-   * @param responseContext
-   * @return
+   * @deprecated use and override {@link #run(QueryPlus, Map)} instead. This method could be removed in the next minor
+   * or major version of Druid.
    */
-  Sequence<T> run(Query<T> query, Map<String, Object> responseContext);
+  @Deprecated
+  default Sequence<T> run(Query<T> query, Map<String, Object> responseContext)
```

Comment:
No implementations. This method is still called in benchmarks and tests at ~270 places.